<launch>

  <include file="$(find mpc)/launch/ekf.launch" />
  <include file="$(find control)/launch/cameraNodeReal.launch" />
  <!-- <rosparam command="load" file="$(find control)/config/gps_config.yaml" /> -->
  <!-- <node pkg="control" type="bridge" name="localization_bridge" output="screen"> </node> -->
  <arg name="sign" default="true"/>
  <arg name="ekf" default="true"/>
  <arg name="lane" default="true"/>
  <arg name="name" default="car1"/>
  <arg name="v" default="18"/>
  <arg name="vref" default="0.$(arg v)"/>
  <!-- <arg name="x0" default="11.71"/>
  <arg name="y0" default="1.896"/>
  <arg name="yaw0" default="0"/> -->
  <arg name="x0" default="0.39"/>
  <arg name="y0" default="5.8"/>
  <arg name="yaw0" default="-1.57"/>
  <arg name="real" default="true"/>
  <arg name="subModel" default="false"/>
  <arg name = "path" default="run1"/>
  <arg name = "rate" default="1000"/>
  <param name="pathName" value="$(arg path)"/>
  <param name="vrefInt" value="$(arg v)"/>
  <arg name = "gmapping" default="false"/>
  <param name="gmapping" value="$(arg gmapping)"/>
  <arg name = "amcl" default="false"/>
  <param name="amcl" value="$(arg amcl)"/>
  <arg name = "lidarOdom" default="false"/>
  <param name="lidarOdom" value="$(arg lidarOdom)"/>
  <arg name = "dashboard" default="true"/>
  <param name="dashboard" value="$(arg dashboard)"/>
  <arg name = "realsense_imu" default="false"/>
  <param name="realsense_imu" value="$(arg realsense_imu)"/>
  <arg name = "kb" default="false"/>
  <param name="kb" value="$(arg kb)"/>
  <arg name = "hw" default="1.33"/>
  <param name="hw" value="$(arg hw)"/>
  <arg name = "gps" default="true"/>
  <param name="gps" value="$(arg gps)"/>
  <arg name = "pub_wpts" default="true"/>
  <param name="pub_wpts" value="$(arg pub_wpts)"/>
  <rosparam command="load" file="$(find mpc)/scripts/config/mpc_config$(arg v).yaml" />
  <rosparam command="load" file="$(find mpc)/config/tunable_params.yaml" />
  <rosparam command="load" file="$(find mpc)/scripts/config/mpc_config_park.yaml" />

  <node pkg="mpc" type="path2.py" name="path_planner" output="screen"></node> 
  <node pkg="mpc" type="controller" name="$(anon mpc_controller)" output="screen"> 
    <param name="sign" value="$(arg sign)"/>
    <param name="ekf" value="$(arg ekf)"/>
    <param name="lane" value="$(arg lane)"/>
    <param name="name" value="$(arg name)"/>
    <param name="/robot_name" value="$(arg name)"/>
    <param name="vref" value="$(arg vref)"/>
    <param name="x0" value="$(arg x0)"/>
    <param name="y0" value="$(arg y0)"/>
    <param name="yaw0" value="$(arg yaw0)"/>
    <param name="real" value="$(arg real)"/>
    <param name="/real" value="$(arg real)"/>
    <param name="subModel" value="$(arg subModel)"/>
    <param name="rate" value="$(arg rate)"/>
  </node>
</launch>
